src/linearmodel.jl

```diff
@@ -104,6 +110,15 @@ acquire_grad_params!(m::LinearACEModel, args...) =
 release_grad_params!(m::LinearACEModel, g) = 
       release_B!(m.basis, g)
 
+
+function ACEbase.valtype(basis::ACEBasis, cfg::AbstractConfiguration, c::AbstractVector{<: SVector})
+   return SVector{length(c[1]), valtype(basis, zero(eltype(cfg)))}
+end
+#calls the regular valtype
+ACEbase.valtype(basis::ACEBasis, cfg::AbstractConfiguration, c::AbstractVector{<: Number}) =
+      valtype(basis, cfg)
+
+
```

Comment:
this confuses me a little. Is it not possible to do this without passing in the parameter vector?

Reply:
This was a way to avoid dispatching several ```acquire_B!``` so we could dispatch there instead. This just seemed cleaner. We could pass c as an argument to ```acquire_B!``` for the dispatching, but otherwise I don't see how to dispatch without c or at least the model.

Reply:
got it, c lives in the model, not in the basis. I still don't quite like it, something feels fishy here. I will just add a comment to revisit this.